File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/direct/Java/C++/s456888268_commented.cpp
#include <bits/stdc++.h>
using namespace std;

void solve() {
    string tmp;
    cin >> tmp;
    int n = tmp.length();
    int ans = 0;
    int cnt = 0;
    for (int i = 0; i < n; i++) {
        if (tmp[i] == 'S') {
            cnt++;
        } else {
            if (cnt == 0) continue;
            ans++;
            cnt--;
        }
    }
    cout << n - 2 * ans << endl;
}

const long long mod = 1e9 + 7;
const int dx[] = { -1, 0, 1, 0 };
const int dy[] = { 0, -1, 0, 1 };
const int dx8[] = { -1, -1, -1, 0, 0, 1, 1, 1 };
const int dy8[] = { -1, 0, 1, -1, 1, -1, 0, 1 };
const int inf = INT_MAX / 2;
const long long linf = LLONG_MAX / 3;
const double dinf = numeric_limits<double>::max() / 3;
const double eps = 1e-10;
const double pi = M_PI;

template <typename T>
void reverse(vector<T>& ar) {
    int len = ar.size();
    for (int i = 0; i < len / 2; i++) {
        swap(ar[i], ar[len - 1 - i]);
    }
}

template <typename T>
void reverse(T ar[], int len) {
    for (int i = 0; i < len / 2; i++) {
        swap(ar[i], ar[len - 1 - i]);
    }
}

string getReverse(string s) {
    reverse(s.begin(), s.end());
    return s;
}

int lowerBound(vector<int>& a, int x) {
    int l = -1, r = a.size();
    while (r - l > 1) {
        int c = (l + r) / 2;
        if (a[c] < x) {
            l = c;
        } else {
            r = c;
        }
    }
    return r;
}

int upperBound(vector<int>& a, int x) {
    int l = -1, r = a.size();
    while (r - l > 1) {
        int c = (l + r) / 2;
        if (a[c] <= x) {
            l = c;
        } else {
            r = c;
        }
    }
    return r;
}

int rlowerBound(vector<int>& a, int x) {
    int l = -1, r = a.size();
    while (r - l > 1) {
        int c = (l + r) / 2;
        if (a[c] > x) {
            l = c;
        } else {
            r = c;
        }
    }
    return r;
}

int rupperBound(vector<int>& a, int x) {
    int l = -1, r = a.size();
    while (r - l > 1) {
        int c = (l + r) / 2;
        if (a[c] >= x) {
            l = c;
        } else {
            r = c;
        }
    }
    return r;
}

int lowerBound(vector<long long>& a, long long x) {
    int l = -1, r = a.size();
    while (r - l > 1) {
        int c = (l + r) / 2;
        if (a[c] < x) {
            l = c;
        } else {
            r = c;
        }
    }
    return r;
}

int upperBound(vector<long long>& a, long long x) {
    int l = -1, r = a.size();
    while (r - l > 1) {
        int c = (l + r) / 2;
        if (a[c] <= x) {
            l = c;
        } else {
            r = c;
        }
    }
    return r;
}

int rlowerBound(vector<long long>& a, long long x) {
    int l = -1, r = a.size();
    while (r - l > 1) {
        int c = (l + r) / 2;
        if (a[c] > x) {
            l = c;
        } else {
            r = c;
        }
    }
    return r;
}

int rupperBound(vector<long long>& a, long long x) {
    int l = -1, r = a.size();
    while (r - l > 1) {
        int c = (l + r) / 2;
        if (a[c] >= x) {
            l = c;
        } else {
            r = c;
        }
    }
    return r;
}

int lowerBound(vector<double>& a, double x) {
    int l = -1, r = a.size();
    while (r - l > 1) {
        int c = (l + r) / 2;
        if (a[c] < x) {
            l = c;
        } else {
            r = c;
        }
    }
    return r;
}

int upperBound(vector<double>& a, double x) {
    int l = -1, r = a.size();
    while (r - l > 1) {
        int c = (l + r) / 2;
        if (a[c] <= x) {
            l = c;
        } else {
            r = c;
        }
    }
    return r;
}

int rlowerBound(vector<double>& a, double x) {
    int l = -1, r = a.size();
    while (r - l > 1) {
        int c = (l + r) / 2;
        if (a[c] > x) {
            l = c;
        } else {
            r = c;
        }
    }
    return r;
}

int rupperBound(vector<double>& a, double x) {
    int l = -1, r = a.size();
    while (r - l > 1) {
        int c = (l + r) / 2;
        if (a[c] >= x) {
            l = c;
        } else {
            r = c;
        }
    }
    return r;
}

int lowerBound(vector<char>& a, char x) {
    int l = -1, r = a.size();
    while (r - l > 1) {
        int c = (l + r) / 2;
        if (a[c] < x) {
            l = c;
        } else {
            r = c;
        }
    }
    return r;
}

int upperBound(vector<char>& a, char x) {
    int l = -1, r = a.size();
    while (r - l > 1) {
        int c = (l + r) / 2;
        if (a[c] <= x) {
            l = c;
        } else {
            r = c;
        }
    }
    return r;
}

int rlowerBound(vector<char>& a, char x) {
    int l = -1, r = a.size();
    while (r - l > 1) {
        int c = (l + r) / 2;
        if (a[c] > x) {
            l = c;
        } else {
            r = c;
        }
    }
    return r;
}

int rupperBound(vector<char>& a, char x) {
    int l = -1, r = a.size();
    while (r - l > 1) {
        int c = (l + r) / 2;
        if (a[c] >= x) {
            l = c;
        } else {
            r = c;
        }
    }
    return r;
}

template <typename T>
int lowerBound(vector<T>& ls, T x) {
    return lower_bound(ls.begin(), ls.end(), x) - ls.begin();
}

template <typename T>
int upperBound(vector<T>& ls, T x) {
    return upper_bound(ls.begin(), ls.end(), x) - ls.begin();
}

template <typename T>
int rupperBound(vector<T>& ls, T x) {
    return upper_bound(ls.rbegin(), ls.rend(), x) - ls.rbegin();
}

template <typename T>
int rlowerBound(vector<T>& ls, T x) {
    return lower_bound(ls.rbegin(), ls.rend(), x) - ls.rbegin();
}

vector<int> concat(int x, vector<int> arr) {
    arr.insert(arr.begin(), x);
    return arr;
}

vector<int> concat(vector<int> arr, int x) {
    arr.push_back(x);
    return arr;
}

vector<long long> concat(long long x, vector<long long> arr) {
    arr.insert(arr.begin(), x);
    return arr;
}

vector<long long> concat(vector<long long> arr, long long x) {
    arr.push_back(x);
    return arr;
}

int max(int x, int y) {
    return std::max(x, y);
}

int min(int x, int y) {
    return std::min(x, y);
}

int max(int x, int y, int z) {
    return std::max(x, std::max(y, z));
}

int min(int x, int y, int z) {
    return std::min(x, std::min(y, z));
}

long long max(long long x, long long y) {
    return std::max(x, y);
}

long long min(long long x, long long y) {
    return std::min(x, y);
}

long long max(long long x, long long y, long long z) {
    return std::max(x, std::max(y, z));
}

long long min(long long x, long long y, long long z) {
    return std::min(x, std::min(y, z));
}

double max(double x, double y) {
    return std::max(x, y);
}

double min(double x, double y) {
    return std::min(x, y);
}

double max(double x, double y, double z) {
    return std::max(x, std::max(y, z));
}

double min(double x, double y, double z) {
    return std::min(x, std::min(y, z));
}

void sort(vector<int>& ar) {
    std::sort(ar.begin(), ar.end());
}

void sort(vector<long long>& ar) {
    std::sort(ar.begin(), ar.end());
}

void sort(vector<double>& ar) {
    std::sort(ar.begin(), ar.end());
}

void sort(vector<char>& ar) {
    std::sort(ar.begin(), ar.end());
}

void rsort(vector<int>& ar) {
    std::sort(ar.rbegin(), ar.rend());
}

void rsort(vector<long long>& ar) {
    std::sort(ar.rbegin(), ar.rend());
}

void rsort(vector<double>& ar) {
    std::sort(ar.rbegin(), ar.rend());
}

void rsort(vector<char>& ar) {
    std::sort(ar.rbegin(), ar.rend());
}

void fill(vector<int>& arr, int x) {
    std::fill(arr.begin(), arr.end(), x);
}

void fill(vector<long long>& arr, long long x) {
    std::fill(arr.begin(), arr.end(), x);
}

void fill(vector<bool>& arr, bool x) {
    std::fill(arr.begin(), arr.end(), x);
}

void fill(vector<double>& arr, double x) {
    std::fill(arr.begin(), arr.end(), x);
}

void fill(vector<vector<int>>& arr, int x) {
    for (auto& row : arr) {
        std::fill(row.begin(), row.end(), x);
    }
}

void fill(vector<vector<long long>>& arr, long long x) {
    for (auto& row : arr) {
        std::fill(row.begin(), row.end(), x);
    }
}

void fill(vector<vector<double>>& arr, double x) {
    for (auto& row : arr) {
        std::fill(row.begin(), row.end(), x);
    }
}

void fill(vector<vector<bool>>& arr, bool x) {
    for (auto& row : arr) {
        std::fill(row.begin(), row.end(), x);
    }
}

long long plus(long long x, long long y) {
    long long res = (x + y) % mod;
    return res < 0 ? res + mod : res;
}

long long sub(long long x, long long y) {
    long long res = (x - y) % mod;
    return res < 0 ? res + mod : res;
}

long long mul(long long x, long long y) {
    long long res = (x * y) % mod;
    return res < 0 ? res + mod : res;
}

long long div(long long x, long long y) {
    return x * pow(y, mod - 2) % mod;
}

long long pow(long long x, long long y) {
    if (y < 0) return 0;
    if (y == 0) return 1;
    if (y % 2 == 1) return (x * pow(x, y - 1)) % mod;
    long long root = pow(x, y / 2);
    return root * root % mod;
}

int main() {
    solve();
    return 0;
}
// End of Code

File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/inline/uncomment/direct/Java/Go/s456888268_commented.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

var (
	mod    = int64(1e9 + 7)
	dx     = []int{-1, 0, 1, 0}
	dy     = []int{0, -1, 0, 1}
	dx8    = []int{-1, -1, -1, 0, 0, 1, 1, 1}
	dy8    = []int{-1, 0, 1, -1, 1, -1, 0, 1}
	inf    = int(^uint(0) >> 1)
	linf   = int64(^uint(64) >> 1)
	dinf   = float64(^uint(64) >> 1)
	eps    = 1e-10
	pi     = math.Pi
	sb     strings.Builder
	is     = bufio.NewReader(os.Stdin)
	out    = bufio.NewWriter(os.Stdout)
	input  string
)

func solve() {
	tmp := ns()
	s := []rune(tmp)
	n := len(s)
	ans := 0
	cnt := 0
	for i := 0; i < n; i++ {
		if s[i] == 'S' {
			cnt++
		} else {
			if cnt == 0 {
				continue
			}
			ans++
			cnt--
		}
	}
	fmt.Fprintln(out, n-2*ans)
}

func reverse[T any](ar []T) {
	length := len(ar)
	for i := 0; i < length/2; i++ {
		ar[i], ar[length-1-i] = ar[length-1-i], ar[i]
	}
}

func getReverse(s string) string {
	c := []rune(s)
	reverse(c)
	return string(c)
}

func lowerBound(a []int, x int) int {
	l, r := -1, len(a)
	for r-l > 1 {
		c := (l + r) / 2
		if a[c] < x {
			l = c
		} else {
			r = c
		}
	}
	return r
}

func upperBound(a []int, x int) int {
	l, r := -1, len(a)
	for r-l > 1 {
		c := (l + r) / 2
		if a[c] <= x {
			l = c
		} else {
			r = c
		}
	}
	return r
}

func rlowerBound(a []int, x int) int {
	l, r := -1, len(a)
	for r-l > 1 {
		c := (l + r) / 2
		if a[c] > x {
			l = c
		} else {
			r = c
		}
	}
	return r
}

func rupperBound(a []int, x int) int {
	l, r := -1, len(a)
	for r-l > 1 {
		c := (l + r) / 2
		if a[c] >= x {
			l = c
		} else {
			r = c
		}
	}
	return r
}

func lowerBound(a []int64, x int64) int {
	l, r := -1, len(a)
	for r-l > 1 {
		c := (l + r) / 2
		if a[c] < x {
			l = c
		} else {
			r = c
		}
	}
	return r
}

func upperBound(a []int64, x int64) int {
	l, r := -1, len(a)
	for r-l > 1 {
		c := (l + r) / 2
		if a[c] <= x {
			l = c
		} else {
			r = c
		}
	}
	return r
}

func rlowerBound(a []int64, x int64) int {
	l, r := -1, len(a)
	for r-l > 1 {
		c := (l + r) / 2
		if a[c] > x {
			l = c
		} else {
			r = c
		}
	}
	return r
}

func rupperBound(a []int64, x int64) int {
	l, r := -1, len(a)
	for r-l > 1 {
		c := (l + r) / 2
		if a[c] >= x {
			l = c
		} else {
			r = c
		}
	}
	return r
}

func lowerBound(a []float64, x float64) int {
	l, r := -1, len(a)
	for r-l > 1 {
		c := (l + r) / 2
		if a[c] < x {
			l = c
		} else {
			r = c
		}
	}
	return r
}

func upperBound(a []float64, x float64) int {
	l, r := -1, len(a)
	for r-l > 1 {
		c := (l + r) / 2
		if a[c] <= x {
			l = c
		} else {
			r = c
		}
	}
	return r
}

func rlowerBound(a []float64, x float64) int {
	l, r := -1, len(a)
	for r-l > 1 {
		c := (l + r) / 2
		if a[c] > x {
			l = c
		} else {
			r = c
		}
	}
	return r
}

func rupperBound(a []float64, x float64) int {
	l, r := -1, len(a)
	for r-l > 1 {
		c := (l + r) / 2
		if a[c] >= x {
			l = c
		} else {
			r = c
		}
	}
	return r
}

func lowerBound(a []rune, x rune) int {
	l, r := -1, len(a)
	for r-l > 1 {
		c := (l + r) / 2
		if a[c] < x {
			l = c
		} else {
			r = c
		}
	}
	return r
}

func upperBound(a []rune, x rune) int {
	l, r := -1, len(a)
	for r-l > 1 {
		c := (l + r) / 2
		if a[c] <= x {
			l = c
		} else {
			r = c
		}
	}
	return r
}

func rlowerBound(a []rune, x rune) int {
	l, r := -1, len(a)
	for r-l > 1 {
		c := (l + r) / 2
		if a[c] > x {
			l = c
		} else {
			r = c
		}
	}
	return r
}

func rupperBound(a []rune, x rune) int {
	l, r := -1, len(a)
	for r-l > 1 {
		c := (l + r) / 2
		if a[c] >= x {
			l = c
		} else {
			r = c
		}
	}
	return r
}

func lowerBound[T any](ls []T, x T) int {
	if len(ls) == 0 {
		return -1
	}
	switch ls[0].(type) {
	case int:
		return ~sort.Search(len(ls), func(i int) bool { return ls[i].(int) >= x.(int) })
	case int64:
		return ~sort.Search(len(ls), func(i int) bool { return ls[i].(int64) >= x.(int64) })
	case float64:
		return ~sort.Search(len(ls), func(i int) bool { return ls[i].(float64) >= x.(float64) })
	default:
		panic("Unsupported type for binary search")
	}
}

func upperBound[T any](ls []T, x T) int {
	if len(ls) == 0 {
		return -1
	}
	switch ls[0].(type) {
	case int:
		return ~sort.Search(len(ls), func(i int) bool { return ls[i].(int) > x.(int) })
	case int64:
		return ~sort.Search(len(ls), func(i int) bool { return ls[i].(int64) > x.(int64) })
	case float64:
		return ~sort.Search(len(ls), func(i int) bool { return ls[i].(float64) > x.(float64) })
	default:
		panic("Unsupported type for binary search")
	}
}

func rupperBound[T any](ls []T, x T) int {
	if len(ls) == 0 {
		return -1
	}
	switch ls[0].(type) {
	case int:
		return ~sort.Search(len(ls), func(i int) bool { return ls[i].(int) < x.(int) })
	case int64:
		return ~sort.Search(len(ls), func(i int) bool { return ls[i].(int64) < x.(int64) })
	case float64:
		return ~sort.Search(len(ls), func(i int) bool { return ls[i].(float64) < x.(float64) })
	default:
		panic("Unsupported type for binary search")
	}
}

func rlowerBound[T any](ls []T, x T) int {
	if len(ls) == 0 {
		return -1
	}
	switch ls[0].(type) {
	case int:
		return ~sort.Search(len(ls), func(i int) bool { return ls[i].(int) <= x.(int) })
	case int64:
		return ~sort.Search(len(ls), func(i int) bool { return ls[i].(int64) <= x.(int64) })
	case float64:
		return ~sort.Search(len(ls), func(i int) bool { return ls[i].(float64) <= x.(float64) })
	default:
		panic("Unsupported type for binary search")
	}
}

func concat(x int, arr []int) []int {
	ret := make([]int, len(arr)+1)
	copy(ret[1:], arr)
	ret[0] = x
	return ret
}

func concat(arr []int, x int) []int {
	ret := make([]int, len(arr)+1)
	copy(ret[:len(ret)-1], arr)
	ret[len(ret)-1] = x
	return ret
}

func concat(x int64, arr []int64) []int64 {
	ret := make([]int64, len(arr)+1)
	copy(ret[1:], arr)
	ret[0] = x
	return ret
}

func concat(arr []int64, x int64) []int64 {
	ret := make([]int64, len(arr)+1)
	copy(ret[:len(ret)-1], arr)
	ret[len(ret)-1] = x
	return ret
}

func max(x int, y int) int {
	return int(math.Max(float64(x), float64(y)))
}

func min(x int, y int) int {
	return int(math.Min(float64(x), float64(y)))
}

func max(x int, y int, z int) int {
	return int(math.Max(float64(x), math.Max(float64(y), float64(z))))
}

func min(x int, y int, z int) int {
	return int(math.Min(float64(x), math.Min(float64(y), float64(z))))
}

func max(x int64, y int64) int64 {
	return int64(math.Max(float64(x), float64(y)))
}

func min(x int64, y int64) int64 {
	return int64(math.Min(float64(x), float64(y)))
}

func max(x int64, y int64, z int64) int64 {
	return int64(math.Max(float64(x), math.Max(float64(y), float64(z))))
}

func min(x int64, y int64, z int64) int64 {
	return int64(math.Min(float64(x), math.Min(float64(y), float64(z))))
}

func max(x float64, y float64) float64 {
	return math.Max(x, y)
}

func min(x float64, y float64) float64 {
	return math.Min(x, y)
}

func max(x float64, y float64, z float64) float64 {
	return math.Max(x, math.Max(y, z))
}

func min(x float64, y float64, z float64) float64 {
	return math.Min(x, math.Min(y, z))
}

func sortInt(ar []int) {
	sort.Ints(ar)
}

func sortInt64(ar []int64) {
	sort.Slice(ar, func(i, j int) bool { return ar[i] < ar[j] })
}

func sortFloat64(ar []float64) {
	sort.Float64s(ar)
}

func sortRune(ar []rune) {
	sort.Slice(ar, func(i, j int) bool { return ar[i] < ar[j] })
}

func rsortInt(ar []int) {
	sort.Ints(ar)
	length := len(ar)
	for i := 0; i < length/2; i++ {
		ar[i], ar[length-1-i] = ar[length-1-i], ar[i]
	}
}

func rsortInt64(ar []int64) {
	sort.Slice(ar, func(i, j int) bool { return ar[i] < ar[j] })
	length := len(ar)
	for i := 0; i < length/2; i++ {
		ar[i], ar[length-1-i] = ar[length-1-i], ar[i]
	}
}

func rsortFloat64(ar []float64) {
	sort.Float64s(ar)
	length := len(ar)
	for i := 0; i < length/2; i++ {
		ar[i], ar[length-1-i] = ar[length-1-i], ar[i]
	}
}

func rsortRune(ar []rune) {
	sort.Slice(ar, func(i, j int) bool { return ar[i] < ar[j] })
	length := len(ar)
	for i := 0; i < length/2; i++ {
		ar[i], ar[length-1-i] = ar[length-1-i], ar[i]
	}
}

func fillInt(arr []int, x int) {
	for i := range arr {
		arr[i] = x
	}
}

func fillInt64(arr []int64, x int64) {
	for i := range arr {
		arr[i] = x
	}
}

func fillBool(arr []bool, x bool) {
	for i := range arr {
		arr[i] = x
	}
}

func fillFloat64(arr []float64, x float64) {
	for i := range arr {
		arr[i] = x
	}
}

func fillIntMatrix(arr [][]int, x int) {
	for i := range arr {
		fillInt(arr[i], x)
	}
}

func fillInt64Matrix(arr [][]int64, x int64) {
	for i := range arr {
		fillInt64(arr[i], x)
	}
}

func fillFloat64Matrix(arr [][]float64, x float64) {
	for i := range arr {
		fillFloat64(arr[i], x)
	}
}

func fillBoolMatrix(arr [][]bool, x bool) {
	for i := range arr {
		fillBool(arr[i], x)
	}
}

func plus(x int64, y int64) int64 {
	res := (x + y) % mod
	if res < 0 {
		res += mod
	}
	return res
}

func sub(x int64, y int64) int64 {
	res := (x - y) % mod
	if res < 0 {
		res += mod
	}
	return res
}

func mul(x int64, y int64) int64 {
	res := (x * y) % mod
	if res < 0 {
		res += mod
	}
	return res
}

func div(x int64, y int64) int64 {
	res := x * pow(y, mod-2) % mod
	if res < 0 {
		res += mod
	}
	return res
}

func pow(x int64, y int64) int64 {
	if y < 0 {
		return 0
	}
	if y == 0 {
		return 1
	}
	if y%2 == 1 {
		return (x * pow(x, y-1)) % mod
	}
	root := pow(x, y/2)
	return root * root % mod
}

func main() {
	defer out.Flush()
	solve()
}

func ns() string {
	line, _ := is.ReadString('\n')
	return strings.TrimSpace(line)
}

func ni() int {
	line, _ := is.ReadString('\n')
	i, _ := strconv.Atoi(strings.TrimSpace(line))
	return i
}

func nl() int64 {
	line, _ := is.ReadString('\n')
	i, _ := strconv.ParseInt(strings.TrimSpace(line), 10, 64)
	return i
}
End of Code.
